sql/analyzer/resolve_functions.go

```diff
@@ -17,9 +17,57 @@ package analyzer
 import (
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/grant_tables"
 	"github.com/dolthub/go-mysql-server/sql/plan"
 )
 
+func resolveTableFunctions(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("resolve_table_functions")
+	defer span.Finish()
+
+	return plan.TransformUp(n, func(n sql.Node) (sql.Node, error) {
+		if n.Resolved() {
+			return n, nil
+		}
+
+		utf, ok := n.(*expression.UnresolvedTableFunction)
+		if !ok {
+			return n, nil
+		}
+
+		tableFunction, err := a.Catalog.TableFunction(ctx, utf.TableFunctionName())
+		if err != nil {
+			return nil, err
+		}
+
+		database, err := a.Catalog.Database(ctx, ctx.GetCurrentDatabase())
```

Comment:
This won't work if there's no current DB selected, or if somebody wants to run a table function that refers to a different DB than the one currently selected.

What would be best here is if table functions had some way to specify tables as arguments. Then they could be fully qualified with the database name as necessary.

I think this is fine for now but longer term we should be moving in that other direction.

Reply:
Good catch. I think I see what you're getting at – could TableFunctions somehow show they are dependent on a specific table and have that table loaded/validated as part of the plan. That feels like it could be pretty handy for table functions that do depend on a real table, but could be optional so other table functions, like JSON_TABLE, didn't have to specify any. I'll follow up with you on this one to see if you see more details in your head.